lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+     */
+    code_verifier?: string
+  }
+
+  export interface IExtendedCallbackChecks extends ICallbackChecks {
+    /**
+     * When provided the authorization response's ID Token auth_time parameter will be checked to be conform to the
+     * max_age value. Use of this check is required if you sent a max_age parameter into an authorization request.
+     */
+    max_age?: number
+    /**
+     * When provided the authorization response's ID Token nonce parameter will be checked to be the this expected
+     * one. Use of this check is required if you sent a nonce parameter into an authorization request.
+     */
+    nonce?: string
+  }
+
+  export interface ICallbackExtras {
+    /**
+     * extra request body properties to be sent to the AS during code exchange.
+     */
+    exchangeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion. This is only used
+     * when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IRefreshExtras {
+    /**
+     * extra request body properties to be sent to the AS during refresh token exchange.
+     */
+    exchangeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IGrantBody {
+    grant_type: string
+    [key: string]: any
+  }
+
+  export interface IGrantExtras {
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IIntrospectExtras {
+    /**
+     * extra request body properties to be sent to the introspection endpoint.
+     */
+    introspectBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload: object
+  }
+
+  export interface IRevokeExtras {
+    /**
+     * extra request body properties to be sent to the revocation endpoint.
+     */
+    revokeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IRequestObjectPayload {
+    client_id?: string
+    iss?: string
+    aud?: string
+    iat?: number
+    exp?: number
+    jti?: string
+    [key: string]: any
```

Comment:
`unknown` is a safer type to use here, so the developer has to explicitly cast it to what they believe the type should be